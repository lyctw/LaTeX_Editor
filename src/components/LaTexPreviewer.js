import React, { useState } from 'react';
import Shortcut from './Shortcut'
import { Button } from '@instructure/ui-buttons'
import { TextArea } from '@instructure/ui-forms'


export default class LaTexPreviewer extends React.Component {
    constructor(props) {
      super(props);
      this.state = {
        latex: ''
      }
    }
  
    handleLatexChange = event => {
      this.setState({latex: event.target.value});
      if(this.state.latex) {
        this.props.enableLatexPreviewer();
      }
      else {
        this.props.disableLatexPreviewer();
      }
    }

    handleAppendTex = (_tex) => {
      this.setState((prevState) => ({latex: prevState.latex + _tex}))
    } 
  
    render() {
      return (
        <div>
          {/* <img src={"https://i.imgur.com/Nbvxf0C.png"} /> */}

          <Shortcut handleAppendTex={this.handleAppendTex}/>
          <br/>
          <TextArea 
            value={this.state.latex} 
            onChange={this.handleLatexChange}
            placeholder="E = mc^2"
            label="Input LaTeX"
          />
          <div id="present-tex">
              <p>Preview:</p>
              <img src={String.raw`https://latex.codecogs.com/svg.latex?\LARGE ${this.state.latex}`} />
          </div>

          <Button 
              onClick={() => { 
                  this.props.handleInsert(this.state.latex);
              }}
          >Add</Button> 
        </div>
      )
    }
};

