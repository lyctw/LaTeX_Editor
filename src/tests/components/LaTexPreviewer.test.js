import React from 'react'
import { shallow } from 'enzyme'
import LaTexPreviewer from '../../components/LaTexPreviewer'
import toJson from 'enzyme-to-json';
 
test('should render LaTexPreviewer', () => {
    const wrapper = shallow(<LaTexPreviewer />);
    expect(toJson(wrapper)).toMatchSnapshot();
});
