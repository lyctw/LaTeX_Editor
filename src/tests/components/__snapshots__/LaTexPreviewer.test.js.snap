// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render LaTexPreviewer 1`] = `
<div>
  <Shortcut
    handleAppendTex={[Function]}
  />
  <br />
  <TextArea
    autoGrow={true}
    disabled={false}
    inline={false}
    label="Input LaTeX"
    layout="stacked"
    messages={Array []}
    onChange={[Function]}
    placeholder="E = mc^2"
    readOnly={false}
    required={false}
    resize="none"
    size="medium"
    textareaRef={[Function]}
    value=""
  />
  <div
    id="present-tex"
  >
    <p>
      Preview:
    </p>
    <img
      src="https://latex.codecogs.com/svg.latex?\\\\LARGE "
    />
  </div>
  <br />
  <Button
    as="button"
    buttonRef={[Function]}
    cursor="pointer"
    fluidWidth={false}
    margin="0"
    onClick={[Function]}
    size="medium"
    type="button"
    variant="default"
  >
    Add
  </Button>
</div>
`;
